### PR TITLE
feat(CLOUDDST-23791): Set signing identity to reference with tag

### PR DESCRIPTION
**Pull request by midnightercz**

### CODE REVIEW

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml [hunk 1]
```diff
@@ -75,22 +76,26 @@ spec:
             if [ $LIST -eq 1 ]; then
                 for PUB_CONTAINER_REF in $PUB_CONTAINER_REFS; do
                     for MDIGEST in $(echo "$IMAGE" | jq -r '.manifests[]|.digest'); do
-                        echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${MDIGEST}"
-                        cosign -t 3m0s sign\
-                        ${COSIGN_COMMON_ARGS}\
-                        --sign-container-identity "${PUB_CONTAINER_REF}@${MDIGEST}"\
-                        "${INTERNAL_CONTAINER_REF}@${MDIGEST}"
+                        for TAG in $TAGS; do
+                            echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${TAG} (${MDIGEST})"
```
Comment (emilyzheng) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml]:
```suggestion
                            echo "Signing manifest ${INTERNAL_CONTAINER_REF}:${TAG} (${MDIGEST})"
```

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml [hunk 2]
```diff
@@ -75,22 +76,26 @@ spec:
...
 
             # Sign manifest list itself or manifest if it's not list
             for PUB_CONTAINER_REF in $PUB_CONTAINER_REFS; do
-                echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${DIGEST}"
-                cosign -t 3m0s sign\
-                ${COSIGN_COMMON_ARGS}\
-                --sign-container-identity "${PUB_CONTAINER_REF}@${DIGEST}"\
-                "${INTERNAL_CONTAINER_REF}@${DIGEST}"
+                for TAG in $TAGS; do
+                    echo "Signing manifest ${INTERNAL_CONTAINER_REF}@${DIGEST}"
```
Comment (emilyzheng) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml]: Include TAG in this output as well?

**Review by johnbieren — COMMENTED** [review]
/ok-to-test
Can you bump the version string in the task yaml and add an entry to the README.md?

**Comment by midnightercz** [comment]
> /ok-to-test Can you bump the version string in the task yaml and add an entry to the README.md?

done

**Review by johnbieren — DISMISSED** [review]

**Comment by johnbieren** [comment]
Needs rebase, but then it will be good to merge

**Comment by johnbieren** [comment]
Looks like the commit also isn't signed. See https://github.com/konflux-ci/release-service/blob/main/CONTRIBUTING.md#signing-commits

**Comment by midnightercz** [comment]
> Looks like the commit also isn't signed. See https://github.com/konflux-ci/release-service/blob/main/CONTRIBUTING.md#signing-commits

Yeah, I didn't realize autorebase performed over ui also needs to be signed

**Comment by johnbieren** [comment]
Still needs commits signed

**Review by johnbieren — DISMISSED** [review]

**Review by johnbieren — DISMISSED** [review]

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml
Comment (mmalina) [inline tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: Question: Shouldn't we sign both the tags and the sha reference? What if the customers pulls the image via the `@sha` reference? Just checking.

Comment (midnightercz) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: If it's pulled by the digest, digest is verified before the signatures is - according to the documentation .I assume by simply running the checksum on the manifest

**Review by johnbieren — DISMISSED** [review]

**Review by johnbieren — APPROVED** [review]

File: tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml
Comment (mmalina) [reply tasks/rh-sign-image-cosign/rh-sign-image-cosign.yaml:1]: I'm not sure I understand. But if you think this use case will work, then it's fine by me.

**Review by mmalina — APPROVED** [review]

**Pull request merged**